src/haml/printer.ts

```diff
@@ -31,7 +31,7 @@ function printHashKey(key: string, opts: Plugin.Options) {
   let quoted = key;
   const joiner = opts.rubyHashLabel ? ":" : " =>";
 
-  if (key.includes(":") || key.includes("-")) {
+  if (key.includes(":") || key.includes("-") || key.includes('@')) {
```

Comment:
Handling the case here avoids touching `Haml::Parser::ParseNode#parse_value`, and looked like a better place. I missed this in the original PR.